onnxruntime/core/providers/xnnpack/xnnpack_execution_provider.h

```diff
@@ -9,13 +9,17 @@
 #include "core/graph/constants.h"
 #include "core/providers/providers.h"
 
+struct pthreadpool;
 namespace onnxruntime {
 // placeholder for future use. no options currently
 struct XnnpackExecutionProviderInfo {
+  int xnn_thread_pool_size{0};
   XnnpackExecutionProviderInfo() = default;
 
-  XnnpackExecutionProviderInfo(const ProviderOptions&) {
-    // future: parse ProviderOptions
+  XnnpackExecutionProviderInfo(const ProviderOptions& po) {
+    if (po.count("intra_op_num_threads")) {
+      xnn_thread_pool_size = std::stoi(po.at("intra_op_num_threads"));
+    }
```

Comment:
nit: if you want to use the entry you're looking up use `find` so there's only one lookup (vs. one lookup for `count` and one for `at`).

Reply:
Make sense with `if statement with initializer`